api/lib/domain/services/schooling-registrations-xml-service.js

```diff
@@ -1,91 +1,216 @@
-  return new DOMParser().parseFromString(stringifiedXml);
+function _extractUAI(path, encoding) {
+  return _withSiecleStream(path, encoding, _extractUAIFromStream);
 }
 
-function _findSchoolingRegistrationStructures(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures) {
-  const xmlSchoolingRegistrationId = xmlSchoolingRegistration.getAttribute('ELEVE_ID');
-  return xmlSchoolingRegistrationsStructures
-    .find((xmlSchoolingRegistrationStructure) => xmlSchoolingRegistrationStructure.getAttribute('ELEVE_ID') === xmlSchoolingRegistrationId);
+async function _processSiecleFile(path, encoding) {
+  return _withSiecleStream(path, encoding, _extractStudentRegistrationsFromStream);
 }
 
-function _findSchoolingRegistrationDivision(schoolingRegistrationStructuresContainer) {
-  if (_.isEmpty(schoolingRegistrationStructuresContainer)) {
-    return null;
+async function _withSiecleStream(path, encoding, fn) {
+  const rawStream = fs.createReadStream(path);
+  const siecleFileStream = rawStream.pipe(iconv.decodeStream(encoding));
+
+  try {
+    return await fn(siecleFileStream);
+  } finally {
+    rawStream.close();
   }
-  const divisionStructureElement = Array.from(schoolingRegistrationStructuresContainer.getElementsByTagName('STRUCTURE'))
-    .find((structureElement) => _getEitherElementValueOrNull(structureElement, 'TYPE_STRUCTURE') === 'D');
+}
+
+function _extractUAIFromStream(siecleFileStream) {
+  return new Promise(function(resolve, reject) {
+    const saxParser = sax.createStream(true);
+    saxParser.on('error', () => {
+      reject(new FileValidationError('XML invalide'));
+    });
+
+    const streamerToParseOrganizationUAI = new saxPath.SaXPath(saxParser, NODE_ORGANIZATION_UAI);
+
+    streamerToParseOrganizationUAI.once('match', (xmlNode) => {
+      xml2js.parseString(xmlNode, (err, nodeData) => {
+        if (err) return reject(err);
+        if (nodeData.PARAMETRES) {
+          const UAIFromSIECLE = _getValueFromParsedElement(nodeData.PARAMETRES.UAJ);
+          resolve(UAIFromSIECLE);
+        }
+      });
+    });
 
-  return divisionStructureElement ? _getEitherElementValueOrNull(divisionStructureElement, 'CODE_STRUCTURE') : null;
+    saxParser.on('end', () => {
+      reject(new FileValidationError('UAI non trouvé dans le fichier'));
+    });
+
+    siecleFileStream.pipe(saxParser);
+  });
 }
 
-function _filterLeftSchoolingRegistrations(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures) {
-  const leavingDate = _getEitherElementValueOrNull(xmlSchoolingRegistration, 'DATE_SORTIE');
-  return _.isEmpty(leavingDate) && _filterSchoolingRegistrationsWithNoDivision(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures);
+function _extractStudentRegistrationsFromStream(siecleFileStream) {
+  return new Promise(function(resolve, reject_) {
+    const saxParser = sax.createStream(true);
+    saxParser.on('error', () => {
+      reject(new FileValidationError('XML invalide'));
```

Comment:
ici il faudrait mettre la même erreur que précédemment : 
```suggestion
      reject(new FileValidationError('Aucun élève n’a pu être importé depuis ce fichier. Vérifiez que le fichier est conforme.''));
```